### DISCUSSION TITLE
Material icon option possibility?

### DISCUSSION THREAD

**Discussion by semi710** [kyazdani42/nvim-web-devicons#281] [Q&A] [answered]
is it possible to have [Material](https://marketplace.visualstudio.com/items?itemName=PKief.material-icon-theme) icons ?
if it is already possible then tell me how to do that?

**Comment by alex-courtis** [comment]
It looks like material icons are present by extension: https://github.com/nvim-tree/nvim-web-devicons/blob/master/lua/nvim-web-devicons.lua#L958

Is there a material `&filetype` ?

**Reply by semi710** [reply]
no not a material filetype
i meant that nvim-web-devicons have specific set of icons.
is there any way to have the material icons as representation ?
like if there any override available ? 
https://marketplace.visualstudio.com/items?itemName=PKief.material-icon-theme

**Reply by gegoune** [reply, marked as answer]
There isn't any other way than manually overriding all icons. We could consider introducing `theme` or `preset`, but that would greatly increase maintenance cost and presets could easily get out of sync (where one preset contains an icon and others don't).

**Reply by semi710** [reply]
will it be possible to have a preset of required icons list that the community can maintain as seprate repo.
and then it can get override in setup config?.

e.g installing xyz/material-icon-for-nvimwebdev

then requiring the material icon and overriding it in here if having a option is possible?

**Reply by alex-courtis** [reply]
That is indeed an option. A separate plugin could setup nvim-web-devicons, passing it overridden icon sets.

**Reply by gegoune** [reply]
There already is https://github.com/yamatsum/nvim-nonicons.

**Comment by gipo355** [comment]
Please make this happen! been trying to hack material icons into neovim for a year now.
Up to now I'm manually overriding but it's quite messy and doesn't work flawlessly.

here is a link to an override list: (https://github.com/DaikyXendo/nvim-material-icon/blob/main/lua/nvim-material-icon.lua)

i've manually copied all the values and overriden them but not everything is working well.

**Comment by mskelton** [comment]
Hey folks, a while back I created a project called [termicons](https://github.com/mskelton/termicons) which basically takes the icon set from the VS Code material icons extension and packages it as an icon font. Combined with [termicons.nvim](https://github.com/mskelton/termicons.nvim), you can use the exact set of vs code icons with nvim-web-devicons!

![image](https://github.com/nvim-tree/nvim-web-devicons/assets/25914066/79527d14-1464-42e6-aa5c-fe1d85ee51b8)


@alex-courtis Not sure if it's something you'd be interested in potentially including as an official configuration. If so, let me know.

**Reply by alex-courtis** [reply]
That's really nice.

An override set similar to https://github.com/yamatsum/nvim-nonicons would be the easiest. Compatibility with nvim-web-devicons wouldn't be an issue as API is used.

Let us know how you go and we'll add an extensions section to the readme or maybe wiki.

**Reply by mskelton** [reply]
@alex-courtis I updated the docs for termicons to include some more instructions. The termicons plugin already is setup in a way very much like how nvim-nonicons is where calling `require('termicons').setup()` does all the work for you.

**Reply by alex-courtis** [reply]
This timing seems pretty fortuitous.

Using a consistent set of seti icons appears achievable, as well as using the names instead of raw values: https://github.com/nvim-tree/nvim-web-devicons/pull/376#issuecomment-1890843503

That will open up some possibilities with an alternative theme. Watch this space...

**Reply by semi710** [reply]
@mskelton i tried termicons nvim it works perfectly.
i am using foot terminal in that i needed to add a line
```
font=JetBrainsMono Nerd Font Mono:size=10,termicons:size=10
```
this sets termicons as fallback. (if possible you can add it in readme)

**Reply by gegoune** [reply]
> this sets termicons as fallback. (if possible you can add it in readme)

I don't think we want to nor have time and knowledge to maintain configurations for all existing terminal emulators out there. User maintained wiki is definitely an option though.

**Reply by mskelton** [reply]
@niksingh710 I agree with @gegoune that maintaining instructions for every single terminal is not something that is very maintainable.

**Reply by semi710** [reply]
A community maintained wiki is a nice option.

**Reply by alex-courtis** [reply]
> A community maintained wiki is a nice option.

It's on the list
